java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -3,21 +3,28 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
 import com.sun.jna.Pointer;
 
+import java.io.IOException;
+
 public class UnleashEngine {
     private static final String UTF_8 = "UTF-8";
     private final YggdrasilFFI yggdrasil;
-    private final ObjectMapper mapper;
+    private final ObjectReader reader;
+    private final ObjectWriter writer;
 
     public UnleashEngine() {
         this(new YggdrasilFFI());
     }
 
     UnleashEngine(YggdrasilFFI yggdrasil) {
         this.yggdrasil = yggdrasil;
-        mapper = new ObjectMapper();
+        ObjectMapper mapper = new ObjectMapper();
         mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        reader = mapper.reader();
+        writer = mapper.writer();
```

Comment:
Following https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance

Reply:
Oh! I didn't know this. I learned something today

Reply:
So did I, this feels like it has changed since I learned Jackson, I see the wiki page was updated in 2019, so definitely long after I learned Jackson. Cool. TIL :)